Add support for Google Cloud Storage repositories (#15517)
crate/crate: docs/sql/statements/create-repository.rst

  The port number of the proxy.

.. _sql-create-repo-gcs:

``gcs``
-------

A ``gcs`` repository stores snapshots on the `Google Cloud Storage`_ service.

Parameters
''''''''''

.. _sql-create-repo-gcs-account:

**bucket**
  | *Type:* ``text``
  | *Required*

  Name of the `Google Cloud Storage`_ bucket used for storing snapshots.
  The bucket must already exist before the repository is created.

  .. NOTE::

>     CrateDB masks this parameter. You cannot query the parameter value from
>     the :ref:`sys.repositories <sys-repositories>` table.

Reviewer: ```suggestion
    This parameter will be masked (shown as ``[xxxxx]``) when querying
    :ref:`sys.repositories <sys-repositories>`.
```

Would be shorter and at the same time explain more concretely what is happening.
Same for the notes below

 Endpoint root url, only used for testing to connect to an alternative storage
 provider.

.. _sql-create-repo-gcs-token_uri:

**token_uri**
 | *Type:* ``text``
 | *Required:* ``false``

 Endpoint oauth token URI, only used for testing to connect to an alternative

>  oauth provider.

Reviewer: Should this be "Only use this for testing" ?  For public documentation it imho doesn't make sense to tell our users how we use this setting.
